packages/wrangler/src/d1/migrations/apply.tsx

```diff
@@ -146,15 +146,10 @@ Your database may not be available to serve requests during the migration, conti
 		}
 
 		for (const migration of unappliedMigrations) {
-			let query = fs.readFileSync(
-				`${migrationsPath}/${migration.name}`,
-				"utf8"
-			);
-			query += `
-								INSERT INTO ${migrationsTableName} (name)
-								values ('${migration.name}');
+			let query = `
```

Comment:
Why is it safe to update the migrations table to say we've completed the migration *before* the migration happens?  Are we relying on transactions somehow?

Reply:
My assumption is that a query string is executed as a single query, so if one of the queries fail, the DO throws... right @geelen ?

Reply:
Wow I guess I missed this at the time (I am cleaning up some old tabs), but yes. Anything submitted in a single query to a D1 is executed in a transaction.